pkg/dhcp6client/dhcp6client.go

```diff
@@ -106,38 +108,40 @@ func (p *buffer) remaining() []byte {
 	return p.consume(len(p.buf))
 }
 
+// If the client fails to receive a valid DHCPv6 packet via socket,
+// it keeps listening until the time is out.
 func (c *Client) ReadReply() (*dhcp6.Packet, error) {
+	tick := time.NewTicker(c.timeout)
+	start := time.Now()
+
 	var err error
-	for i := 0; i < 5; i++ {
-		// set a random timeout within range [c.timeout, 2*c.timeout)
-		// so as to prevent swamping a DHCP server.
-		c.connection.SetReadTimeout(c.timeout + time.Duration(rand.Intn(int(c.timeout))))
-		pb := make([]byte, 1500)
-		var n int
-		n, err = c.connection.ReadFrom(pb)
-		if err != nil {
-			// Return error if read time of socket is due,
-			// so that request can be resent instantly
-			if err, ok := err.(net.Error); ok && err.Timeout() {
-				return nil, err
+	for {
+		select {
+		case <-tick.C:
+			return nil, err
```

Comment:
This error would be nil if we never receive any packet.

Also, you may not want to retry unless a timeout happened, so you should define a retry error type:

```
var ErrReadTimedOut = errors.New("waiting for response timed out")

...
if timeout <= 0 {
  return 0, ErrReadTimedOut
}

...
packet, err = c.ReadReply()
if err == ErrReadTimedOut {
  continue
}
```

But maybe we should retry regardless of whether it was a timeout error? Either way, this err could be nil.